README.md
# YSPDO
Class in PHP `YSPDO` is a `PDO` helper to manipulate dynamically database records using `arrays`
### Initialize class

```php
$yspdo = new YSPDO([
  'mysql',  // Driver
  'host'    => 'localhost',
  'port'    => 3306,
  'dbname'  => 'generatedata'
],'root','');
```
## CREATE | READ | UPDATE | DELETE
#### CREATE
```php
$stmt = $yspdo->insert('peoples',[
  'name'    => 'Gabriel Almeida',
  'email'   => 'gabrieel@email.com',
  'country' => 'Brazil'
]);

echo $stmt->lastInsertId();
```
#### READ
```php
$stmt = $yspdo->select('peoples',['name','email','address','company'],[
  'id{>=}' => 100
]);

print_r( $stmt->fetchAll('COLUMN') );
```
#### UPDATE
```php
$stmt = $yspdo->update('peoples',[
  'phone' => '(00) 0000-0000',
  'address' => '4129 Magna. Avenue',
],[
  'id' => 100
]);
```
#### DELETE
```php
$stmt = $yspdo->delete('peoples',[
  'id' => 100
]);
```
## FUNCTION SELECT
#### fetch
```php
$yspdo->select('peoples','*',[
    'email' => 'commodo@sem.edu'
])->fetch();
```
#### fetchAll
```php
$yspdo->select('peoples')->fetchAll();
```
#### fetch setting type of return
```php
$yspdo->select('peoples','*',[
    'email' => 'commodo@sem.edu'
])->fetch('OBJ');
```

#### fetch all, defining type of return
```php
$yspdo->select('peoples')->fetchAll('OBJ');
```

#### Selecting columns
```php
$yspdo->select('peoples',['name','email','phone'])->fetchAll();
```
#### DISTINCT
```php
$yspdo->select('peoples',[
    'DISTINCT' => 'name'
    // OR
    'DISTINCT' => ['name','phone','date']
])->fetchAll();
```

#### ALIASES
```php
$yspdo->select('peoples',[
    'AS' => [
        'name' => 'yourName',
        'date' => 'birthday',
        'email' => 'contact'
    ]
])->fetchAll();
```

#### Operators
```php
$yspdo->select('peoples','*',[
    'id{>=}' => 100
])->fetchAll();
```

#### rowCount
```php
$yspdo->select('peoples','*',[
    'email' => 'dolor.sit@ametdiam.ca'
])->rowCount();
```
#### ORDER BY
```php
$yspdo->select('peoples','*',[
    'ORDER' => 'city'
    // OR
    'ORDER' => 'name DESC'
    // OR
    'ORDER' => ['name DESC','city ASC']
])->fetchAll();
```
#### BETWEEN
```php
$yspdo->select('peoples','*',[
    'BETWEEN' => [
        'id' => [1,25]
        // OR
        'name' => ['a','b']
    ]
])->fetchAll();
```
#### NOT BETWEEN
```php
$yspdo->select('peoples','*',[
    '!BETWEEN' => [
        'id' => [1,25]
        // OR
        'name' => ['a','b']
    ]
])->fetchAll();
```
#### LIKE
```php
$yspdo->select('peoples','*',[
    'LIKE' => [
        'name' => 'g%'
    ]
])->fetchAll();
```
#### NOT LIKE
```php
$yspdo->select('peoples','*',[
    '!LIKE' => [
        'name' => 'g%'
    ]
])->fetchAll();
```
#### LIMIT
```php
$yspdo->select('peoples','*',[
    'LIMIT' => 10
])->fetchAll()
```
#### IN
```php
$yspdo->select('peoples','*',[
    'IN' => [
        'city' => ['Acoz','Pietraroja','Martelange','Relegem']
    ]
])->fetchAll();
```
#### NOT IN
```php
$yspdo->select('peoples','*',[
    '!IN' => [
        'city' => ['Acoz','Pietraroja','Martelange','Relegem']
    ]
])->fetchAll();
```
#### Using IN | BETWEEN | NOT LIKE | ORDER BY
```php
$yspdo->select('peoples','*',[
    'IN' => [
        'date' => ['02/11/1985','11/27/1997','09/24/1969','01/15/1985','09/12/1986']
    ],
    'BETWEEN' => [
        'id' => [1,50]
    ],
    '!LIKE' => [
        'city' => 's%'
    ],
    'ORDER' => 'city ASC'
])->fetchAll();
```
### CREATE DATABASE
```php
$yspdo->createDB('dbname');
```
### DELETE DATABASE
```php
$yspdo->deleteDB('dbname');
```
### INSERT INTO
```php
$yspdo->insert('peoples',[
    'id'        => 301,
    'name'      => 'Gabriel',
    'phone'     => '+0000000000',
    'date'      => '00/00/0000',
    'email'     => 'name@test.com',
    'address'   => 'Street teste address nº 1000 Av test',
    'city'      => 'an_city',
    'country'   => 'Brazil',
    'company'   => 'No Company',
]);
```
### UPDATE
```php
$yspdo->update('peoples',[
    'phone' => '+1111111111',
    'date' => '11/11/1111',
    'email' => 'name@server.co',
    'address' => 'Street teste address nº 1000 Av test'
],[ 'id' => 301 ]);
```
### DELETE
```php
$yspdo->delete('peoples',[
    'id' => 301
]);
```
### COUNT
```php
$yspdo->count('peoples','*',[
    'id{>}' => 125
]);
```
### DELETE TABLE
```php
$yspdo->deleteTable('peoples');
```
### TRUNCATE TABLE
```php
$yspdo->truncate('peoples');
```
***
### Attention
#### Do not use the (select|insert|update|delete|count|prepare) functions like this:
```php
$statement1 = $yspdo->(select|insert|update|delete|count|prepare)('table_one' ...);
$statement2 = $yspdo->(select|insert|update|delete|count|prepare)('table_two' ...);

print_r( $statement1->fetchAll() );
print_r( $statement2->fetchAll() );

// YSPDO will replace the query from $statement1 with $statement2
```
***
### Available Functions
```php
/**
* PDO::getAvailableDrivers
*
* @return array
*/
->getAvailableDrivers()
/**
* PDO::commit
*
* @return boolean
*/
->commit()
/**
* PDO::beginTransaction
*
* @return boolean
*/
->beginTransaction()
/**
* PDO::rollBack
*
* @return boolean
*/
->rollBack()
/**
* PDO::inTransaction
*
* @return boolean
*/
->inTransaction()
/**
* PDO::exec
*
* @param string $statement
* @return integer|boolean
*/
->exec( string $statement )
/**
* PDO::quote
*
* @param string $string
* @param int $parameter_type
* @return string
*/
->quote($string [, $parameter_type=\PDO::PARAM_STR])
/**
* PDO::errorCode
*
* @return mixed
*/
->errorCode()
/**
* PDO::errorInfo
*
* @return array
*/
->errorInfo()
/**
* Query Statement
*
* @param string $sql
* @param array $parameters
* @return YSPDO
*/
->query(string $sql [, array $parameters=[]])
/**
* Row count
*
* @return integer
*/
->rowCount()
/**
* Last insert id
*
* @return integer
*/
->lastInsertId()
/**
* Prepare Statement
*
* @param string $sql
* @return YSPDO
*/
->prepare(string $sql)
/**
* bindColumn Statement
*
* @param mixed $column
* @param mixed $param
* @param int $type
* @param int $maxlen
* @param mixed $driverdata
* @return YSPDO
*/
->bindColumn(mixed $column , mixed &$param [, int $type [, int $maxlen [, mixed $driverdata ]]])
/**
* bindParam Statement
*
* @param mixed $parameter
* @param mixed $value
* @param int $type
* @param int $length
* @return YSPDO
*/
->bindParam(mixed $parameter , mixed &$variable [, int $data_type = PDO::PARAM_STR [, int $length [, mixed $driver_options ]]])
/**
* bindValue Statement
*
* @param mixed $parameter
* @param mixed $value
* @param int $type
* @return YSPDO
*/
->bindValue(mixed $parameter , mixed $value [, int $data_type = PDO::PARAM_STR ])
/**
* closeCursor Statement
*
* @return boolean
*/
->closeCursor()
/**
* columnCount Statement
*
* @return int
*/
->columnCount()
/**
* execure Statement
*
* @param array $parameters
* @return boolean
*/
->execute([array $parameters=[]])
/**
* fetch Statement
*
* @param string $style
* @param int $cursor_orientation
* @param int $offset
* @return mixed
*/
->fetch([ string $style [, int $cursor_orientation = PDO::FETCH_ORI_NEXT [, int $cursor_offset = 0 ]]])
/**
* fetchAll Statement
*
* @param string $style
* @param mixed $argument
* @param array $ctor_args
* @return mixed
*/
->fetchAll([ string $style [, mixed $argument = \PDO::FETCH_COLUMN [, array $ctor_args = [] ]]])
/**
* fetchColumn Statement
*
* @param int $column_number
* @return midex
*/
->fetchColumn([int $column_number = 0])
/**
* fetchObject Statement
*
* @param string $class_name
* @param array $ctor_args
* @return midex
*/
->fetchObject([ string $class_name = "stdClass" [, array $ctor_args = [] ]])
/**
* getAttribute Statement
*
* @param int $attribute
* @return mixed
*/
->getAttribute(int $attribute)
/**
* setAttribute Statement
*
* @param int $attribute
* @param mixed $value
* @return boolean
*/
->setAttribute(int $attribute , mixed $value)
/**
* SELECT
*
* @param string       $table
* @param string|array $columns
* @param array|null   $where
* @return YSPDO
*/
->select(string $table [,$columns='*' [,$where=null]])
/**
* INSERT
*
* @param string $table
* @param array  $data
* @return YSPDO
*/
->insert(string $table, array $data)
/**
* UPDATE
*
* @param string $table
* @param array  $data
* @param array|null  $where
* @return YSPDO
*/
->update(string $table, array $data [, $where=null])
/**
* DELETE
*
* @param string $table
* @param array  $where
* @return YSPDO
*/
->delete(string $table [, array $where=''])
/**
*  COUNT
*
* @param string $table
* @param string|array $columns
* @param array $where
* @return string
*/
->count(string $table [, $columns = '*' [, array $where = []]])
/**
* Create Database
*
* @param string $s Database name
* @return boolean
*/
->createDB(string $s)
/**
* Drop Database
*
* @param string $s Database name
* @return boolean
*/
->deleteDB(string $s)
/**
* Drop Table
*
* @param string $s Table name
* @return boolean
*/
->deleteTable(string $s)
/**
* Truncate Table
*
* @param string $s Table name
* @return boolean
*/
->truncate(string $s)
```
***
## REQUIREMENTS
PHP Version 7.x or newer
## LICENCE
[MIT](/LICENCE.txt)


CHANGELOG.md
# Change Log

## 1.1.0 / 2017-03-04
### Added
- Function: count
- (private) Function: err_exception

### Changed
- Function rename: '_backtick' ~ 'backtick', and code improvements
- Function rename: '_cWhere' ~ 'cWHERE', and code improvements
- Code improvements in function: cSQL

### Removed
- Support for prefix and suffix
- Support for selecting all columns with 'all'
- Function: createTable
- (private) Function: _PDOException
- File: examples\select.php

## 1.0.1 / 2016-09-22
### Added
- File CHANGELOG.md
- File examples/connections.php
- Prefix and suffix for columns names
- Function _buildDSN()

### Changed
- Connection with drivers PDO are now generic

### Removed
- Function _setSettings()
- Function init()
- File dist/YSPDO-Structure.txt


examples/select.php
<?php

include __DIR__ . "/../dist/YSPDO.php";


$db = new YSPDO([
  'mysql',
  'host' => 'localhost',
  'port' => 3306,
  'dbname' => 'generatedata',
  'charset' => 'utf8',
],'root','');


/**
* YSPDO->select( $table, $columns, $where )
*
  * @param string       $table
  * @param string|array $columns  [optional]
  * @param array        $where    [optional]
  * @return object [class YSPDO]
*
*/


/**
   Remove the comments and enjoy it
**/


// Column
//
// $query = $db->select('peoples',['name']);
// Or without array (recommended)
// $query = $db->select('peoples','name');
//
// SQL Statement: SELECT `name` FROM `peoples`;


// Columns
//
// $query = $db->select('peoples',['name','email','address','company']);
// Or without array
// $query = $db->select('peoples','`name`,`email`,`address`,`company`');
//
// SQL Statement: SELECT `name`,`email`,`address`,`company` FROM `peoples`;


// DISTINCT
//
// $query = $db->select( 'peoples' ,[ 'DISTINCT' => '`name`' ]);
// Or using array
// $query = $db->select( 'peoples' ,[ 'DISTINCT' => ['name','phone','date'] ]);
// Without array
// $query = $db->select( 'peoples' , 'DISTINCT `name`,`phone`,`date`');
//
// SQL Statement: SELECT DISTINCT `name`,`phone`,`date` FROM `peoples`;


// ALIASES
//
// $query = $db->select( 'peoples' ,[
//   'AS' => [
//     'name' => 'yourName',
//     'date' => 'birthday',
//     'email' => 'contact'
//   ]
// ]);
// Without array
// $query = $db->select( 'peoples' , '`name` AS `yourName`,`date` AS `birthday`,`email` AS `contact`');
//
// SQL Statement: SELECT `name` AS `yourName`,`date` AS `birthday`,`email` AS `contact` FROM `peoples`;


// COMPARISON OPERATORS
//
// $query = $db->select( 'peoples' ,'*', [ 'id{>=}' => 10, 'id{<=}' => 50 ]);
// Without array
// $query = $db->select( 'peoples' ,'*', 'WHERE id>=10 AND id<=50');
//
// SQL Statement (with array): SELECT * FROM `peoples` WHERE id>=? AND id<=?;
// SQL Statement (without array): SELECT * FROM `peoples` WHERE id>=10 AND id<=50;


// ORDER BY
//
// $query = $db->select( 'peoples' ,'all',[
//   // 'ORDER' => 'city'
//   // OR
//   // 'ORDER' => 'name DESC'
//   // OR
//   'ORDER' => ['name DESC','city ASC']
// ]);
// Without array
// $query = $db->select( 'peoples' ,'all', 'ORDER BY `name` DESC, `city` ASC');
//
// SQL Statement: SELECT * FROM `peoples` ORDER BY `name` DESC, `city` ASC;


// BETWEEN
//
// $query = $db->select( 'peoples' ,'all',[
//   'BETWEEN' => [
//     'id' => [1,275] // accepts only array with two values being only string or number
//   ]
// ]);
// Without array
// $query = $db->select( 'peoples' , 'all', 'WHERE `id` BETWEEN 1 AND 275');
//
// SQL Statement: SELECT * FROM `peoples` WHERE `id` BETWEEN 1 AND 275;


// NOT BETWEEN
//
// $query = $db->select( 'peoples' ,'all',[
//   '!BETWEEN' => [
//     'name' => ['a','d'] // accepts only array with two values being only string or number
//   ]
// ]);
// Without array
// $query = $db->select( 'peoples' , 'all', 'WHERE `name` NOT BETWEEN \'a\' AND \'d\'');
//
// SQL Statement: SELECT * FROM `peoples` WHERE `name` NOT BETWEEN 'a' AND 'd';


// NOT LIKE
//
// $query = $db->select( 'peoples' ,'all',[
//   '!LIKE' => [
//     'name' => 'c%'
//   ]
// ]);
// Without array
// $query = $db->select( 'peoples' , 'all', 'WHERE `name` NOT LIKE \'c%\'');
//
// SQL Statement: SELECT * FROM `peoples` WHERE `name` NOT LIKE 'c%';


// LIKE
//
// $query = $db->select( 'peoples' ,'*',[
//   'LIKE' => [
//     'city' => '%es%'
//   ]
// ]);
// Without array
// $query = $db->select( 'peoples' , 'all', 'WHERE `city` LIKE \'%es%\'');
//
// SQL Statement: SELECT * FROM `peoples` WHERE `city` LIKE '%es%';


// LIMIT
//
// $query = $db->select( 'peoples' ,'ALL', [ 'LIMIT' => 10 ]);
// Without array
// $query = $db->select( 'peoples' ,'ALL', 'LIMIT 10');
//
// SQL Statement: SELECT * FROM `peoples` LIMIT 10;


// IN
//
// $query = $db->select( 'peoples' ,'ALL',[
//   'IN' => [
//     'city' => ['Acoz','Pietraroja','Martelange','Relegem']
//   ]
// ]);
// Without array
// $query = $db->select( 'peoples' ,'ALL', 'WHERE `city` IN (\'Acoz\',\'Pietraroja\',\'Martelange\',\'Relegem\')');
//
// SQL Statement: SELECT * FROM `peoples` WHERE `city` IN ('Acoz','Pietraroja','Martelange','Relegem');


// NOT IN
//
// $query = $db->select( 'peoples' ,'all',[
//   '!IN' => [
//     'city' => ['Acoz','Pietraroja','Martelange','Relegem']
//   ]
// ]);
// Without array
// $query = $db->select( 'peoples' ,'all', 'WHERE `city` NOT IN (\'Acoz\',\'Pietraroja\',\'Martelange\',\'Relegem\')');
//
// SQL Statement: SELECT * FROM `peoples` WHERE `city` NOT IN ('Acoz','Pietraroja','Martelange','Relegem');


print_r( $query->fetchAll() );


examples/connections.php
<?php die('see the source of this file');


// CUBRID
$db = new YSPDO([
  'cubrid',
  'dbname' => 'demodb',
  'host' => '192.168.0.10',
  'port' => 30000
],'username','password');
// DSN cubrid:dbname=demodb;host=192.168.0.10;port=30000



// MS SQL Server
$db = new YSPDO([
  'mssql',
  'host' => 'sqlserver01.database.com',
  'dbname' => 'db'
],'username','password');
// DSN mssql:host=sqlserver01.database.com;dbname=db



// Firebird
$db = new YSPDO([
  'firebird',

  // PDO_FIREBIRD DSN example with path
  'dbname' => '/path/to/DATABASE.FDB',

  // or PDO_FIREBIRD DSN example with port and path
  'dbname' => 'hostname/port:/path/to/DATABASE.FDB',

  // or PDO_FIREBIRD DSN example with localhost and path to employee.fdb on Debian system
  'dbname' => 'localhost:/var/lib/firebird/2.5/data/employee.fdb'
]);



// IBM
$db = new YSPDO([
  'ibm',
  'DRIVER'    => '{IBM DB2 ODBC DRIVER}',
  'DATABASE'  => 'testdb',
  'HOSTNAME'  => '11.22.33.444',
  'PORT'      => '56789',
  'PROTOCOL'  => 'TCPIP'
],'username','password');
// DSN ibm:DRIVER={IBM DB2 ODBC DRIVER};DATABASE=testdb;HOSTNAME=11.22.33.444;PORT=56789;PROTOCOL=TCPIP



// INFORMIX
$db = new YSPDO([
  'informix',
  'host'                    => 'host.domain.com',
  'service'                 => '9800',
  'database'                => 'common_db',
  'server'                  => 'ids_server',
  'protocol'                => 'onsoctcp',
  'EnableScrollableCursors' => '1'
],'username','password');
// DSN informix:host=host.domain.com;service=9800;database=common_db;server=ids_server;protocol=onsoctcp;EnableScrollableCursors=1



// MySQL (support functions: select|insert|update|delete|count)
$db = new YSPDO([
  'mysql',
  'host' => 'localhost',
  'dbname' => 'generatedata',
  'port' => 3306,
  'charset' => 'utf8',
],'username','password');
// DSN mysql:host=localhost;dbname=generatedata;port=3306;charset=utf8



// MS SQL Server
$db = new YSPDO([
  'sqlsrv',
  'Server'    => '12345abcde.database.windows.net',
  'Database'  => 'testdb'
],'username','password');
// DSN sqlsrv:Server=12345abcde.database.windows.net;Database=testdb



// Oracle
$db = new YSPDO([
  'oci',

  // Connect to a database defined in tnsnames.ora // DSN oci:dbname=mydb
  'dbname' => 'mydb',

  // or Connect using the Oracle Instant Client // DSN oci:dbname=//localhost:1521/mydb
  'dbname' => '//localhost:1521/mydb'

]);



// ODBC and DB2
$db = new YSPDO(['odbc','MSSQLServer'],'username','password');
// DSN odbc:MSSQLServer



// PostgreSQL
$db = new YSPDO([
  'pgsql',
  'host'      => 'localhost',
  'port'      => 5432,
  'dbname'    => 'testdb',
  'user'      => 'username',
  'password'  => 'password'
]);
// DSN pgsql:host=localhost;port=5432;dbname=testdb;user=bruce;password=mypass



// SQLite
$db = new YSPDO(['sqlite','/opt/databases/mydb.sq3']);
// OR
$db = new YSPDO(['sqlite',':memory:']);
// OR
$db = new YSPDO(['sqlite2','/opt/databases/mydb.sq2']);
// OR
$db = new YSPDO(['sqlite2',':memory:']);



// 4D
$db = new YSPDO([
  '4D',
  'host' => 'localhost',
  'charset' => 'UTF-8'
]);
// DSN 4D:host=localhost;charset=UTF-8



// Only tested on driver MySQL


examples/basic.php
<?php


include __DIR__."/../dist/YSPDO.php";


$db = new YSPDO([
  'mysql',
  'host' => 'localhost',
  'dbname' => 'generatedata',
  'port' => 3306,
  'charset' => 'utf8',
],'root','');



$table = 'peoples';


// Selection all columns
//
// $query = $db->select( $table )->fetchAll();


// Selecting columns
//
$query = $db->select( $table ,['name','email','phone'])->fetchAll();


// Fetch
//
// $query = $db->select( $table ,'*',[
//   'email' => 'commodo@sem.edu'
// ])->fetch();

// Fetch all
//
// $query = $db->select( $table )->fetchAll();
//
// Fetch, with type of return
//
// $query = $db->select( $table ,'*',[
//   'email' => 'commodo@sem.edu'
// ])->fetch('OBJ');
//
// Fetch all, with type of return
//
// $query = $db->select( $table )->fetchAll('OBJ');


// DISTINCT
//
// $query = $db->select( $table ,[
//   // 'DISTINCT' => 'name'
//   // OR
//   // 'DISTINCT' => ['name','phone','date']
// ])->fetchAll();


// ALIASES
//
// $query = $db->select( $table ,[
//   'AS' => [
//     'name' => 'yourName',
//     'date' => 'birthday',
//     'email' => 'contact'
//   ]
// ])->fetchAll();


// Operators
//
// $query = $db->select( $table ,'*',[
//   'id{>=}' => 100
// ])->fetchAll();



// Row count
//
// $query = $db->select( $table ,'*',[
//   'email' => 'dolor.sit@ametdiam.ca'
// ])->rowCount();


// ORDER BY
//
// $query = $db->select( $table ,'*',[
//   // 'ORDER' => 'city'
//   // OR
//   // 'ORDER' => 'name DESC'
//   // OR
//   // 'ORDER' => ['name DESC','city ASC']
// ])->fetchAll();


// NOT BETWEEN
//
// $query = $db->select( $table ,'*',[
//   '!BETWEEN' => [
//     'id' => [1,275]
//     // OR
//     // 'name' => ['a','b']
//   ]
// ])->fetchAll();


// BETWEEN
//
// $query = $db->select( $table ,'*',[
//   'BETWEEN' => [
//     // 'id' => [1,25]
//     // OR
//     // 'name' => ['a','b']
//   ]
// ])->fetchAll();


// NOT LIKE
//
// $query = $db->select( $table ,'*',[
//   '!LIKE' => [
//     'name' => 'g%'
//   ]
// ])->fetchAll();

// LIKE
//
// $query = $db->select( $table ,'*',[
//   'LIKE' => [
//     'city' => '%es%'
//   ]
// ])->fetchAll();


// LIMIT
//
// $query = $db->select( $table ,'*',[
//   'LIMIT' => 10
// ])->fetchAll();


// IN
//
// $query = $db->select( $table ,'*',[
//   'IN' => [
//     'city' => ['Acoz','Pietraroja','Martelange','Relegem']
//   ]
// ])->fetchAll();

// NOT IN
//
// $query = $db->select( $table ,'*',[
//   '!IN' => [
//     'city' => ['Acoz','Pietraroja','Martelange','Relegem']
//   ]
// ])->fetchAll();



// IN | BETWEEN | NOT LIKE | ORDER BY
//
// $query = $db->select( $table ,'*',[
//   'IN' => [
//     'city' => ['Zeist','Stafford','Remagne','Istres','Warwick','Urbe','Chatillon','Rawalpindi']
//   ],
//   'BETWEEN' => [
//     'id' => [1,250]
//   ],
//   '!LIKE' => [
//     'city' => 's%'
//   ],
//   'ORDER' => 'city ASC'
// ])->fetchAll();


// CREATE DATABASE
//
// $query = $db->createDB('dbname');


// DELETE DATABASE
//
// $query = $db->deleteDB('dbname');


// INSERT INTO
//
// $query = $db->insert( $table ,[
//
//   'id' => 301,
//   'name' => 'Gabriel',
//   'phone' => '+0000000000',
//   'date' => '00/00/0000',
//   'email' => 'name@test.com',
//   'address' => 'Street teste address nº 1000 Av test',
//   'city' => 'Araguaina',
//   'country' => 'Brazil',
//   'company' => 'No Company',
//
// ]);


// UPDATE
//
// $query = $db->update( $table ,[
//
//   'phone' => '+1111111111',
//   'date' => '11/11/1111',
//   'email' => 'name@server.co',
//   'address' => 'Street teste address nº 1000 Av test'
//
// ],[ 'id' => 301 ]);


// DELETE
//
// $query = $db->delete( $table ,[
//   'id' => 301
// ]);
//
// Delete all data (Use ->rowCount() to know how many rows have been affected)
// $query = $db->delete( $table ); 
// OR 
// $query = $db->delete( $table , '*' );


// COUNT
// 
// $query = $db->count( $table, '*', [
//   // 'LIKE' => [
//   //   'address' => '%Avenue%'
//   // ]
//   // 
//   // 'id{>}' => 275
// ]);
// 
// var_dump( $query );


// DELETE TABLE
//
// $query = $db->deleteTable( $table );


// TRUNCATE TABLE
//
// $query = $db->truncate( $table );


print_r( $query );
